tests/fixtures/MakeAuthenticatorLoginFormUserEntity/tests/SecurityControllerTest.php

```diff
@@ -60,7 +61,17 @@ public function testCommand()
         $client->submit($form);
 
         $this->assertStringContainsString('TODO: provide a valid redirect', $client->getResponse()->getContent());
-        $this->assertNotNull($token = $client->getContainer()->get('security.token_storage')->getToken());
+
+        // Handle Session deprecations in Symfony 5.3+
+        if (Kernel::VERSION_ID >= 50300) {
+            $tokenStorage = static::$container->get('security.token_storage');
```

Comment:
Why is the way that we fetch the service different in the two cases? This new way seems more correct.

And, would it be possible to use a `method_exists()` instead of the Symfony version?

```php
$tokenStorage = static::$container->get('security.token_storage');
if (method_exists($tokenStorage, 'disableUsageTracking')) {
    $tokenStorage->disableUsageTracking();
}
$token = $tokenStorage->getToken();
```

Checking the kernel is sort of a "last resort". If we can rely on something hard in the code, that's even better.